sotodlib/hwp/update_hwp_angle.py

```diff
@@ -0,0 +1,517 @@
+import os
+import numpy as np
+import scipy.interpolate
+import so3g
+from spt3g import core
+import argparse
+
+class update_hwp_angle(): 
+    
+    def __init__(self, debug=False):
+
+        """
+        Class to manage a L2 HK data into HWP angle g3.
+
+        Args
+        -----
+            debug: bool
+                bit for dubug mode
+        """
+
+        
+        self._debug = debug
+        self._start = 0
+        self._end = 0
+        # Size of pakcets sent from the BBB
+        self._pkt_size = 120 # maybe 120 in the latest version, 150 in the previous version
+        # Number of encoder slits per HWP revolution
+        self._num_edges = 570*2
+        self._delta_angle = 2 * np.pi / self._num_edges
+        self._ref_edges = 2
+        self._ref_indexes = []
+
+    def load_data(self, start, end, archive_path, instance='HBA1'):
+
+        """
+        Loads house keeping data for a given time range. 
+        For the specified time range, this function returns HWP parameters in HK g3 file
+
+        Args
+        -----
+            start : timestamp or DateTime (timezone: UTC)
+                start time for data
+            end :  timestamp  or DateTime (timezone: UTC)
+                end time for data
+            archive_path : str
+                path to HK g3 file
+            instance : str          
+                default = HBA1
+        """
+        self._start = start
+        self._end = end
+        hwp_keys=[
```

Comment:
These need to be in a config file somewhere. `load_range` takes config files as well as fields/aliases. We don't want to have to use different versions of sotodlib if someone changes a feed name. for this function and `load_file` as well.

Reply:
Implemented to work with config yaml file. I put my config file here: `simons1:/mnt/so1/users/ysakurai/hwp_configs/hwp_config.yaml`
But I made it also work without a config file for convenience.